anker/tests/tests/amm.rs

```diff
@@ -0,0 +1,34 @@
+use lido::token::Lamports;
+use solana_program::program_pack::Pack;
+use solana_program_test::tokio;
+use solana_sdk::signer::Signer;
+use testlib::anker_context::Context;
+
+#[tokio::test]
+async fn test_successful_token_swap() {
+    let mut context = Context::new_with_initialized_token_pool().await;
+    let (st_sol_kp, st_sol_token) = context
+        .solido_context
+        .deposit(Lamports(10_000_000_000))
+        .await;
+
+    let ust_address = context
+        .solido_context
+        .create_spl_token_account(
+            context.token_pool_context.ust_mint_address,
+            st_sol_kp.pubkey(),
+        )
+        .await;
+
+    context
+        .swap_st_sol_for_ust(&st_sol_token, &ust_address, &st_sol_kp, 1_000_000_000, 0)
```

Comment:
I only notice this now, but this function takes a naked `u64` for `amount_in` and `minimum_amount_out`. We can make it more self-documenting and harder to misuse by using the stronger types `StLamports` and `UstLamports`: that makes it impossible to accidentally confuse the two.

Also, right now I have to jump to the definition of `swap_st_sol_for_ust` to understand the meaning of the last two arguments. (I can guess that the 1_000_000_000 is the amount in, but I wasn’t sure about the 0 until I looked it up.) We can make it more self-documenting by giving them names:

```rust
let amount_in = StLamports(1_000_000_000);
let min_amount_out = UstLamports(0);
context.swap_st_sol_for_ust(..., amount_in, min_amount_out).await;
```